src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4901,12 +4901,14 @@ private void VisitDeconstructionArguments(ArrayBuilder<DeconstructionVariable> v
                     else
                     {
                         var targetType = variable.Type;
+                        bool leftShouldReInfer = (variable.Expression as BoundLocal)?.DeclarationKind == BoundLocalDeclarationKind.WithInferredType;
+
                         TypeWithState operandType;
                         TypeWithState valueType;
                         int valueSlot;
                         if (underlyingConversion.IsIdentity)
                         {
-                            operandType = default;
+                            operandType = leftShouldReInfer ? VisitRvalueWithState(rightPart) : default;
```

Comment:
>VisitRvalueWithState(rightPart) [](start = 62, length = 31)

We're visiting `rightPart` twice (once here, and once in `VisitOptionalImplicitConversion`) so any warnings will be repeated.

Reply:
Actually, if we're inferring the type of the LHS, we shouldn't be calling `VisitOptionalImplicitConversion` (or we should pass `targetTypeOpt: null` to that method). See `VisitLocalDeclaration` for similar behavior.

---
In reply to: [286779251](https://github.com/dotnet/roslyn/pull/35576#discussion_r286779251) [](ancestors = 286779251)

Reply:
I changed this, so that we don't visit optional implicit conversion when inferring the LHS as we can just use the visited operand type.

---
In reply to: [286783552](https://github.com/dotnet/roslyn/pull/35576#discussion_r286783552) [](ancestors = 286783552,286779251)